test/parallel/xyk-pallet.API.poolLiqudity.multipleUserRewards.test.ts

```diff
@@ -0,0 +1,202 @@
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(2500000);
+process.env.NODE_ENV = "test";
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+let testUser: User;
+let testUser1: User;
+let testUser2: User;
+let sudo: User;
+let keyring: Keyring;
+let token1: BN;
+let token2: BN;
+let liqIdPromPool: BN;
+let mangata: Mangata;
+const defaultCurrencyValue = new BN(250000);
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  keyring = new Keyring({ type: "sr25519" });
+
+  const { chainUri } = getEnvironmentRequiredVars();
+  mangata = await getMangataInstance(chainUri);
+
+  // setup users
+  sudo = new User(keyring, sudoUserName);
+
+  [testUser, testUser1, testUser2] = setupUsers();
+
+  await setupApi();
+
+  [token1, token2] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [defaultCurrencyValue, defaultCurrencyValue],
+    sudo
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.FinalizeTge(),
+    Assets.initIssuance(),
+    Assets.mintToken(token1, testUser, Assets.DEFAULT_AMOUNT),
+    Assets.mintToken(token2, testUser, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser, Assets.DEFAULT_AMOUNT.muln(2)),
+    Assets.mintToken(token1, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintToken(token2, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser1),
+    Assets.mintToken(token1, testUser2, Assets.DEFAULT_AMOUNT),
+    Assets.mintToken(token2, testUser2, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser2),
+    Sudo.sudoAs(
+      testUser,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        Assets.DEFAULT_AMOUNT.divn(2),
+        token1,
+        Assets.DEFAULT_AMOUNT.divn(2)
+      )
+    ),
+    Sudo.sudoAs(
+      testUser,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        Assets.DEFAULT_AMOUNT.divn(2),
+        token2,
+        Assets.DEFAULT_AMOUNT.divn(2)
+      )
+    )
+  );
+
+  testUser1.addAsset(liqIdPromPool);
+  testUser1.addAsset(token1);
+  testUser2.addAsset(liqIdPromPool);
+  testUser2.addAsset(token1);
+});
+
+test("Users minted a different number of tokens THEN they receive an equivalent amount of rewards", async () => {
+  await promotePool(token1);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(
+        MGA_ASSET_ID,
+        token1,
+        defaultCurrencyValue.mul(new BN(2))
+      )
+    ),
+    Sudo.sudoAs(
+      testUser2,
+      Xyk.mintLiquidity(MGA_ASSET_ID, token1, defaultCurrencyValue)
+    )
+  );
+
+  await waitForRewards(testUser1, liqIdPromPool);
+
+  const rewardsAmountUser1 = await mangata.calculateRewardsAmount(
+    testUser1.keyRingPair.address,
+    liqIdPromPool.toString()
+  );
+
+  const rewardsAmountUser2 = await mangata.calculateRewardsAmount(
+    testUser2.keyRingPair.address,
+    liqIdPromPool.toString()
+  );
+
+  expect(rewardsAmountUser1.mul(new BN(2))).bnLte(rewardsAmountUser2);
```

Comment:
Plz expect that is > than zero. 
Why lte? perhaps we can be a bit more accurate? like the diff is less than 1000 ?